### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by simonedavico**
First implementation of the client library.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,38 +1,58 @@
...
+    	<project.name>benchflow-faban-client</project.name>
+    	<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    	<java.version>1.8</java.version>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.httpcomponents</groupId>
+            <artifactId>httpclient</artifactId>
+            <version>4.5.1</version>
```
Comment (VincenzoFerme) [inline pom.xml]: Write the version as a property and reference to it in the dependencies block.

File: pom.xml [hunk 2]
```diff
@@ -1,38 +1,58 @@
...
+        </dependency>
+
+        <dependency>
+            <groupId>org.apache.httpcomponents</groupId>
+            <artifactId>httpmime</artifactId>
+            <version>4.5.1</version>
```
Comment (VincenzoFerme) [inline pom.xml]: Write the version as a property and reference to it in the dependencies block.

File: src/main/java/cloud/benchflow/fabanclient/FabanClient.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
+package cloud.benchflow.fabanclient;
+
+import cloud.benchflow.fabanclient.commands.DeployCommand;
+import cloud.benchflow.fabanclient.configurations.*;
+import cloud.benchflow.fabanclient.exceptions.DeployException;
+import cloud.benchflow.fabanclient.exceptions.FabanClientException;
+import cloud.benchflow.fabanclient.exceptions.JarFileNotFoundException;
+import cloud.benchflow.fabanclient.responses.DeployStatus;
+import com.sun.istack.internal.NotNull;
```
Comment (VincenzoFerme) [inline src/main/java/cloud/benchflow/fabanclient/FabanClient.java]: You may want to use: http://docs.oracle.com/javaee/7/api/javax/validation/constraints/NotNull.html

File: src/main/java/cloud/benchflow/fabanclient/Main.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
+package cloud.benchflow.fabanclient;
```
Comment (VincenzoFerme) [inline src/main/java/cloud/benchflow/fabanclient/Main.java]: This class, or similar, should be a test resource

Comment (simonedavico) [inline src/main/java/cloud/benchflow/fabanclient/Main.java]: It will disappear eventually. It's here now because I wanted a very quick test.

**Comment by VincenzoFerme** [comment]
The package should be cloud.benchflow.faban.client

File: src/main/java/cloud/benchflow/faban/client/Main.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
+package cloud.benchflow.faban.client;
```
Comment (VincenzoFerme) [inline src/main/java/cloud/benchflow/faban/client/Main.java]: This code should be moved in a test package

Comment (simonedavico) [inline src/main/java/cloud/benchflow/faban/client/Main.java]: Moved.

**Pull request merged**